src/Model.php

```diff
@@ -1238,7 +1238,9 @@ public function tryLoadAny()
         $this->data = $this->persistence->tryLoadAny($this);
         if ($this->data) {
             if ($this->id_field) {
-                $this->id = $this->data[$this->id_field];
+                if (isset($this->data[$this->id_field])) {
```

Comment:
I think it's better to use `if (array_key_exists($this->id_field, $this->data))` here in case we actually have id=0 in some strange use-cases.

Reply:
i was actually considering it, but decided against it.